extensions/amp-analytics/0.1/events.js

```diff
@@ -410,6 +525,140 @@ export class IniLoadTracker extends EventTracker {
 }
 
 
+/**
+ * Timer event handler.
+ */
+class TimerEventHandler {
+  /**
+   * @param {JsonObject} timerSpec The timer specification.
+   * @param {function(): UnlistenDef=} startBuilder Factory for building start
+   *     trackers for this timer.
+   * @param {function(): UnlistenDef=} stopBuilder Factory for building stop
+   *     trackers for this timer.
+   */
+  constructor(timerSpec, startBuilder, stopBuilder) {
+    /** {number|undefined} */
+    this.intervalId = undefined;
+
+    /** @const {number} */
+    this.intervalLength = Number(timerSpec['interval']) || 0;
+    user().assert('interval' in timerSpec,
+        'Timer interval specification required');
+    user().assert(this.intervalLength >= MIN_TIMER_INTERVAL_SECONDS,
+        'Bad timer interval specification');
+
+    /** @const {number} */
+    this.maxTimerLength = 'maxTimerLength' in timerSpec ?
+        Number(timerSpec['maxTimerLength']) : DEFAULT_MAX_TIMER_LENGTH_SECONDS;
+    user().assert(this.maxTimerLength > 0, 'Bad maxTimerLength specification');
+
+    /** @const {boolean} */
+    this.callImmediate = 'immediate' in timerSpec ?
+        Boolean(timerSpec['immediate']) : true;
+
+    /** @const {boolean} */
+    this.isUnstoppable = !stopBuilder;
+
+    /** {?UnlistenDef} */
+    this.unlistenStart = null;
+
+    /** {?UnlistenDef} */
+    this.unlistenStop = null;
+
+    /** @const {function(): UnlistenDef|undefined} */
+    this.startBuilder = startBuilder;
+
+    /** @const {function(): UnlistenDef|undefined} */
+    this.stopBuilder = stopBuilder;
+  }
+
+  /**
+   * @param {function()} startTimer
+   */
+  init(startTimer) {
+    if (!this.startBuilder) {
+      // Timer starts on load.
+      startTimer();
+    } else {
+      // Timer starts on event.
+      this.listenForStart();
+    }
+  }
+
+  listenForStart() {
```

Comment:
Regarding to all `!!`. It's purely a code style thing, but as far as I know we don't use that a lot. I used to add !! to covert boolean types, and was asked to remove them : ) 
So please only add `!!` if you are assigning the value to another boolean type value : ) Thanks

Reply:
Done.